src/components/Avatar.tsx

```diff
@@ -0,0 +1,38 @@
+import { Avatar as MuiAvatar, AvatarProps as MuiAvatarProps, Tooltip } from '@mui/material';
+import { useQuery } from '@tanstack/react-query';
+import { fetchUser } from '../api/roleApi';
+import { getInitials } from '../utils/general-helpers';
+import { getFullName } from '../utils/user-helpers';
+
+interface AvatarProps extends Pick<MuiAvatarProps, 'sx'> {
+  username: string;
+}
+
+export const Avatar = ({ username, sx }: AvatarProps) => {
+  const userQuery = useQuery({
+    enabled: !!username,
+    queryKey: ['user', username],
+    queryFn: () => fetchUser(username),
+    retry: 0,
+    staleTime: Infinity,
+    gcTime: 1_800_000, // 30 minutes
+  });
+  const user = userQuery.data;
```

Comment:
Maybe move this to a hook if we're starting with that now?

Reply:
Godt poeng. Will do!